src/CloudFoundry.CloudController.Test.Integration/LoggregatorTest.cs

```diff
@@ -0,0 +1,220 @@
+            }
+
+            PagedResponseCollection<ListAllSpacesResponse> spaces = client.Spaces.ListAllSpaces().Result;
+
+            Guid spaceGuid = Guid.Empty;
+
+            foreach (ListAllSpacesResponse space in spaces)
+            {
+                spaceGuid = space.EntityMetadata.Guid;
+                break;
+            }
+
+            if (spaceGuid == Guid.Empty)
+            {
+                throw new Exception("No spaces found");
+            }
+
+            PagedResponseCollection<ListAllAppsResponse> apps = client.Apps.ListAllApps().Result;
+
+            foreach (ListAllAppsResponse app in apps)
+            {
+                if (app.Name.StartsWith("logTest"))
+                {
+                    client.Apps.DeleteApp(app.EntityMetadata.Guid).Wait();
+                    break;
+                }
+            }
+
+            apprequest = new CreateAppRequest();
+            apprequest.Name = "logTest" + Guid.NewGuid().ToString();
+            apprequest.Memory = 64;
+            apprequest.Instances = 1;
+            apprequest.SpaceGuid = spaceGuid;
+            apprequest.Buildpack = "https://github.com/ryandotsmith/null-buildpack.git";
+            apprequest.EnvironmentJson = new Dictionary<string, string>() { { "envtest1234", "envtestvalue1234" } };
+            apprequest.Command = "printenv; cat content.txt; ping 127.0.0.1;";
+
+            client.Apps.PushProgress += Apps_PushProgress;
+
+            File.WriteAllText(Path.Combine(tempAppPath, "content.txt"), "dummy content");
+        }
+
+        static void Apps_PushProgress(object sender, PushProgressEventArgs e)
+        {
+            Console.WriteLine(e.Message + " " + e.Percent);
+        }
+
+        [TestMethod]
+        [TestCategory("RequiresLoggregator")]
+        public void LoggregatorRecentTest()
+        {
+            CreateAppResponse app = client.Apps.CreateApp(apprequest).Result;
+
+            Guid appGuid = app.EntityMetadata.Guid;
+
+            client.Apps.Push(appGuid, tempAppPath, true).Wait();
+
+            while (true)
+            {
+                var appSummary = client.Apps.GetAppSummary(appGuid).Result;
+                var packageState = appSummary.PackageState.ToLowerInvariant();
+
+                if (packageState != "pending")
+                {
+                    Assert.AreEqual(packageState, "staged");
+
+                    var instances = client.Apps.GetInstanceInformationForStartedApp(appGuid).Result;
+
+                    if (instances.Count > 0)
+                    {
+                        if (instances[0].State.ToLower() == "running")
+                        {
+                            break;
+                        }
+                    }
+                }
+            }
+
+            if (client.Info.GetInfo().Result.LoggingEndpoint == null)
+            {
+                Assert.Inconclusive("CloudFoundry target does not have a loggregator endpoint");
```

Comment:
We've had issues with inconclusive - I remember using categories to decide which tests we want to run against a certain target?